testing/src/scenario/context.py

```diff
@@ -365,6 +365,50 @@ def action(
             kwargs["id"] = id
         return _Event(f"{name}_action", action=_Action(name, **kwargs))
 
+    @staticmethod
+    def custom(
+        event: ops.BoundEvent,
+        *args: Any,
+        **kwargs: Any,
+    ):
+        """Event triggered by a charm library.
+
+        For example, for a library that emits a ``HappenedEvent`` event via a
+        ``MyConsumer`` object, and charm code like::
```

Comment:
I found this slightly difficult to parse at first. How about this (also removing "via"):

```suggestion
        For example, suppose that a library uses a ``MyConsumer`` object to emit an
        ``ItHappened`` event, with charm code::
```